src/workerd/api/sockets.c++

```diff
@@ -57,9 +74,28 @@ jsg::Ref<Socket> connectImplNoOutputLock(
   auto headers = kj::heap<kj::HttpHeaders>(ioContext.getHeaderTable());
   auto httpClient = kj::newHttpClient(*client);
   auto request = httpClient->connect(address, *headers);
+  auto socketStream = kj::mv(request.connection);
+  KJ_IF_MAYBE(opts, options) {
+    if (opts->tls) {
+      // Configure a default TLS context.
+      kj::TlsContext::Options options;
+      options.useSystemTrustStore = true;
+
+      auto tls = kj::heap<kj::TlsContext>(kj::mv(options));
+      auto maybeHost = getHost(address);
+      // TODO(later): Allow custom host to be specified in Socket options.
+      KJ_IF_MAYBE(host, maybeHost) {
+        // Wrap the socket stream in the TLS context.
+        socketStream = kj::newPromisedStream(tls->wrapClient(kj::mv(socketStream), *host));
```

Comment:
I'm not that familiar with kj tls ... does the `tls` instance need to be attached to anything to make sure it stays alive or is that covered by the call to `wrapClient(...)` here?

Reply:
Good point. Looks like other code attaches it.